fix: fix issue with csv stream

tachyonicClock opened the pull request:
* Annapoorna found an issue: if you didn't want to use the final column of a CSV as the target, it wouldn't work. This PR fixes this issue.
* Refactors `Schema.from_custom`
* Rewrites the `CSVStream`. 

    My main issue with the CSV stream is that it uses:
    ```python
        def next_instance(self) -> _AnyInstance:
            if not self.has_more_instances():
                return None
            # skip header
            data = np.genfromtxt(
                self.csv_file_path,
                delimiter=self.delimiter,
                dtype=self.dtypes,
                names=None,
                skip_header=self.n_lines_to_skip,
                max_rows=1,
            )
    ```
    which loads the CSV file for every instance and then parses `self.n_lines_to_skip`. This is inefficient. The old version reads the fried CSV in 30+ seconds, and the new version does it in 1.3s.

* I remove `stream_from_file` as well because the ARFFStream and CSVStream initialization now have very different constructors. You have to specify possible values for categorical data in CSV streams but not ARFF. 

* Removed `ConcatStream` since it is unused.

hmgomes commented:
can we keep `stream_from_file` as deprecated and internally it just maps to the new CSV and ARFF reader functions?

tachyonicClock commented:
I added back ``stream_from_file`` for CSV. It will automatically infer categorical features:
<img width="566" height="236" alt="image" src="https://github.com/user-attachments/assets/a830c55c-51ed-4e44-8923-0388398a1802" />

Updated the tests:
* Ensure missing values are supported.
* Ensure any column can be used as the target.
* Ensure consistent representation and behavior regardless of stream type.

hmgomes commented:
Please check the conflicts, @tachyonicClock
